egs/ifnenit/v1/local/data_prep.sh

```diff
@@ -0,0 +1,142 @@
+  rm tmp.unsorted tmp.sorted
+
+  local/process_data.py $database data/$set --dataset $set --model_type word || exit 1
+  sort data/$set/images.scp -o data/$set/images.scp
+  sort data/$set/utt2spk -o data/$set/utt2spk
+
+  export LC_ALL=C
+  ./utils/utt2spk_to_spk2utt.pl data/$set/utt2spk > data/$set/spk2utt
+  export LC_ALL=$oldLC
+
+  mkdir -p data/{train,test}/data
+
+  local/make_feature_vect.py data/$set --scale-size 40 | \
+    copy-feats --compress=true --compression-method=7 \
+    ark:- ark,scp:data/$set/data/images.ark,data/$set/feats.scp || exit 1
+
+  steps/compute_cmvn_stats.sh data/$set || exit 1;
+
+
+done
+
+
+
+if [ -d "data/local" ]; then
+  rm -r data/local
+fi
+
+## Determine phoneme set
+mkdir -p data/local/lm
+cat data/train/text | cut -d' ' -f2- | tr ' ' "\n" | sort -u > data/local/lm/train.vocab
+cat data/local/lm/train.vocab | python3 local/get_atb_pronun.py > data/train/words2latin
+cat data/train/text | cut -d' ' -f2- | python3 local/rollout_pronuns.py data/train/words2latin | cut -d' ' -f2- | tr ' ' "\n" | sort | uniq -c | awk '{if ($1 > 50 || length($2) == 3) print $2}' | fgrep -v '~A' > data/local/phonemeset
+
+## Lexicon and word/phoneme lists
+mkdir -p data/lang/
+mkdir -p data/local/dict
+echo '<unk>' > data/lang/oov.txt
+cat data/train/words2latin | python3 local/map_to_rareA.py data/local/phonemeset > data/local/dict/lexicon.txt
+echo "<unk> rareA" >> data/local/dict/lexicon.txt
+echo "!SIL sil" >> data/local/dict/lexicon.txt
+export LC_ALL=C
+cat data/local/phonemeset | fgrep -v '.A' | fgrep -v ',A' | fgrep -v 'conn' | fgrep -v 'sil' | sort > data/local/dict/nonsilence_phones.txt
+export LC_ALL=$oldLC
+echo ',A' > data/local/dict/silence_phones.txt
+echo '.A' >> data/local/dict/silence_phones.txt
+echo 'conn' >> data/local/dict/silence_phones.txt
+echo 'rareA' >> data/local/dict/silence_phones.txt
+echo 'sil' >> data/local/dict/silence_phones.txt
+echo 'sil' > data/local/dict/optional_silence.txt
+# config folder
+cat config/extra_questions.org.txt| python3 local/reduce-to-vocabulary.py data/local/dict/nonsilence_phones.txt | sort -u | fgrep ' ' > data/local/dict/extra_questions.txt
+
+export LC_ALL=C
+utils/prepare_lang.sh --num-sil-states 3 --num-nonsil-states 4 --position-dependent-phones false data/local/dict "<unk>" data/local/lang data/lang_pregdl_nolm
+export LC_ALL=$oldLC
+
+
+cut -d ' ' -f 2- data/train/text > data/local/lm/train.lines
+cat data/test/text | awk '{ for(i=2;i<=NF;i++) print $i;}' | sort -u >test_words.txt
+cat data/train/text | awk '{ for(i=2;i<=NF;i++) print $i;}' | sort -u >train_words.txt
+utils/filter_scp.pl --exclude train_words.txt test_words.txt > diff.txt
+cat diff.txt >> data/local/lm/train.lines
```

Comment:
please remove this line (and any other code that might leak any part of the test data into the LM or lexicon). I guess these lines were used for the initial tests and debugging.

Reply:
And move all the code related to LM training to local/train_lm.sh or maybe run.sh